R/Utilities.R

```diff
@@ -124,27 +124,58 @@ function(pm,ki,byrow)
 # -----------------
 #  4. # pMsub     - sub-matrix of a polynomial-matrix
 
-pMsub <-    
-function(pm,i,j=i) 
-  { pd<-polyMconvert(pm,"polyMdlist")
+pMsub <- function(pm, i, j=i) 
+{
+  pd <- polyMconvert(pm, "polyMdlist")
+  
+  if (length(i) > 0) {
+    i <- i[i!=0]
+    i <- i[i<=pd$dim[1]]
     
-    if(length(i)>0) 
-      { i<-i[i!=0];i<-i[i<=pd$dim[1]] 
-	    if(length(i)>0) {
-        if((!all(i<0))&(!all(i>0))) stop("The row selection indices have different signs!")
-        pd$dlist<-pd$dlist[i] # retain or delet the i.th row(s)
-        pd$dim[1] <- if(i[1]<0) pd$dim[1]-length(i) else length(i)  } }
-    if(length(j)>0) 
-      { j<-j[j!=0];j<-j[j<=pd$dim[2]] 
-	    if(length(j)>0) {
-        if((!all(j<0))&(!all(j>0))) stop("The column selection indices have different signs!")
-        for(k in 1:length(pd$dlist)) 
-          pd$dlist[[k]]<-pd$dlist[[k]][j]# retain or delet  the j.th column from the k.th row
-        pd$dim[2] <- if(j[1]<0) pd$dim[2]-length(j) else length(j)  } }
-    pd$degree<-pd$degree[if(length(i)) i else 1:pd$dim[1],if(length(j)) j else 1:pd$dim[2],drop=FALSE] 
-    pm<-polyMconvert(pd,class(pm)[1])		
-    return(pm) 
-   }
+	  if (length(i) > 0) {
+      if ((!all(i < 0)) & (!all(i > 0))) {
+        stop("The row selection indices have different signs!")
+      }
+	    
+      pd$dlist <- pd$dlist[i] # retain or delete the i.th row(s)
+      
+      if (i[1] < 0) {
+        pd$dim[1] <- pd$dim[1]-length(i)
+      } else {
+        pd$dim[1] <- length(i)
+      }
+    } 
+  }
+  
+  if (length(j) > 0) { 
+    j <- j[j!=0]
+    j <- j[j<=pd$dim[2]]
+    
+	  if (length(j) > 0) {

```

Comment:
tabs?

Reply:
shouldn't I use tab inside if?

Reply:
this line contains a mix of space and tabs. You have to use only spaces.